### ISSUE TITLE
Call to a member function fire() on null after upgrading to 1.9

### ISSUE THREAD

**Issue by hackel** [kristijanhusak/laravel-form-builder#289]
I just upgraded from 1.7.10 to 1.9.0 and now I get the following error message when I attempt to render a form.  I have not attempted to debug this yet.
```
[2016-10-06 21:00:34] prod.ERROR: Symfony\Component\Debug\Exception\FatalThrowableError: Call to a member function fire() on null in /vendor/kris/laravel-form-builder/src/Kris/LaravelFormBuilder/Form.php:174
Stack trace:
#0 /vendor/kris/laravel-form-builder/src/Kris/LaravelFormBuilder/Form.php(196): Kris\LaravelFormBuilder\Form->makeField('company[overrid...', 'checkbox', Array)
#1 /app/Http/Forms/CompanyForm.php(29): Kris\LaravelFormBuilder\Form->add('company[overrid...', 'checkbox', Array)
#2 /app/Providers/FormServiceProvider.php(105): App\Http\Forms\CompanyForm->buildForm()
#3 [internal function]: App\Providers\FormServiceProvider->App\Providers\{closure}(Object(Illuminate\View\View))
#4 /bootstrap/cache/compiled.php(9784): call_user_func_array(Object(Closure), Array)
#5 /bootstrap/cache/compiled.php(14216): Illuminate\Events\Dispatcher->fire('composing: admi...', Array)
#6 /bootstrap/cache/compiled.php(14609): Illuminate\View\Factory->callComposer(Object(Illuminate\View\View))
#7 /bootstrap/cache/compiled.php(14594): Illuminate\View\View->renderContents()
#8 /bootstrap/cache/compiled.php(14629): Illuminate\View\View->render()
#9 /bootstrap/cache/compiled.php(14622): Illuminate\View\View->gatherData()
#10 /bootstrap/cache/compiled.php(14610): Illuminate\View\View->getContents()
#11 /bootstrap/cache/compiled.php(14594): Illuminate\View\View->renderContents()
#12 /bootstrap/cache/compiled.php(15756): Illuminate\View\View->render()
#13 /bootstrap/cache/compiled.php(15271): Illuminate\Http\Response->setContent(Object(Illuminate\View\View))
#14 /bootstrap/cache/compiled.php(7717): Symfony\Component\HttpFoundation\Response->__construct(Object(Illuminate\View\View))
#15 /bootstrap/cache/compiled.php(8949): Illuminate\Routing\Router->prepareResponse(Object(App\Http\Request), Object(Illuminate\View\View))
#16 [internal function]: Illuminate\Routing\ControllerDispatcher->Illuminate\Routing\{closure}(Object(App\Http\Request))
#17 /bootstrap/cache/compiled.php(9620): call_user_func(Object(Closure), Object(App\Http\Request))
#18 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#19 /bootstrap/cache/compiled.php(9602): call_user_func(Object(Closure), Object(App\Http\Request))
#20 /bootstrap/cache/compiled.php(8950): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#21 /bootstrap/cache/compiled.php(8935): Illuminate\Routing\ControllerDispatcher->callWithinStack(Object(App\Http\Controllers\Admin\CompanyController), Object(Illuminate\Routing\Route), Object(App\Http\Request), 'edit')
#22 /bootstrap/cache/compiled.php(7884): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(App\Http\Request), 'App\\Http\\Con...', 'edit')
#23 /bootstrap/cache/compiled.php(7855): Illuminate\Routing\Route->runWithCustomDispatcher(Object(App\Http\Request))
#24 /bootstrap/cache/compiled.php(7508): Illuminate\Routing\Route->run(Object(App\Http\Request))
#25 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(App\Http\Request))
#26 /bootstrap/cache/compiled.php(9620): call_user_func(Object(Closure), Object(App\Http\Request))
#27 /app/Http/Middleware/Auth/User.php(48): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#28 [internal function]: App\Http\Middleware\Auth\User->handle(Object(App\Http\Request), Object(Closure), 'admin')
#29 /bootstrap/cache/compiled.php(9612): call_user_func_array(Array, Array)
#30 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#31 /bootstrap/cache/compiled.php(9602): call_user_func(Object(Closure), Object(App\Http\Request))
#32 /bootstrap/cache/compiled.php(7509): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#33 /bootstrap/cache/compiled.php(7497): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(App\Http\Request))
#34 /bootstrap/cache/compiled.php(7482): Illuminate\Routing\Router->dispatchToRoute(Object(App\Http\Request))
#35 /bootstrap/cache/compiled.php(2310): Illuminate\Routing\Router->dispatch(Object(App\Http\Request))
#36 [internal function]: Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(App\Http\Request))
#37 /bootstrap/cache/compiled.php(9620): call_user_func(Object(Closure), Object(App\Http\Request))
#38 /bootstrap/cache/compiled.php(2925): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#39 /app/Http/Middleware/VerifyCsrfToken.php(30): Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(App\Http\Request), Object(Closure))
#40 [internal function]: App\Http\Middleware\VerifyCsrfToken->handle(Object(App\Http\Request), Object(Closure))
#41 /bootstrap/cache/compiled.php(9612): call_user_func_array(Array, Array)
#42 /bootstrap/cache/compiled.php(12971): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#43 [internal function]: Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(App\Http\Request), Object(Closure))
#44 /bootstrap/cache/compiled.php(9612): call_user_func_array(Array, Array)
#45 /bootstrap/cache/compiled.php(11563): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#46 [internal function]: Illuminate\Session\Middleware\StartSession->handle(Object(App\Http\Request), Object(Closure))
#47 /bootstrap/cache/compiled.php(9612): call_user_func_array(Array, Array)
#48 /bootstrap/cache/compiled.php(12708): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#49 [internal function]: Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(App\Http\Request), Object(Closure))
#50 /bootstrap/cache/compiled.php(9612): call_user_func_array(Array, Array)
#51 /bootstrap/cache/compiled.php(12645): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#52 [internal function]: Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(App\Http\Request), Object(Closure))
#53 /bootstrap/cache/compiled.php(9612): call_user_func_array(Array, Array)
#54 /bootstrap/cache/compiled.php(2982): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#55 [internal function]: Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(App\Http\Request), Object(Closure))
#56 /bootstrap/cache/compiled.php(9612): call_user_func_array(Array, Array)
#57 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Http\Request))
#58 /bootstrap/cache/compiled.php(9602): call_user_func(Object(Closure), Object(App\Http\Request))
#59 /bootstrap/cache/compiled.php(2257): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#60 /bootstrap/cache/compiled.php(2240): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(App\Http\Request))
#61 /public/index.php(52): Illuminate\Foundation\Http\Kernel->handle(Object(App\Http\Request))
#62 {main}
```

**Comment by hackel** [comment]
Nevermind, this was caused by 4e2fd1df95fc4c4933f81df4b8eac4edb6a58c0b.  I needed to call `setEventDispatcher` manually when creating my form, which I do manually in a service provider instead of using `FormBuilder::create` so that I can inject the forms directly into my controller methods.

Wouldn't it make more sense to inject the EventDispatcher in the parent Form class instead of setting it this way?
